kitsune/wiki/facets.py

```diff
@@ -73,62 +75,43 @@ def documents_for(locale, topics=None, products=None):
 def _documents_for(locale, topics=None, products=None):
     """Returns a list of articles that apply to passed in topics and products."""
     # First try to get the results from the cache
-    documents = cache.get(_documents_for_cache_key(locale, topics, products))
+    cache_key = _cache_key(locale, topics, products)
+    documents = cache.get(f"documents_for:{cache_key}")
     if documents:
         return documents
 
-    try:
-        # Then try ES
-        documents = _es_documents_for(locale, topics, products)
-        cache.add(_documents_for_cache_key(locale, topics, products), documents)
-    except TransportError:
-        # Finally, hit the database (through cache machine)
-        # NOTE: The documents will be the same ones returned by ES
-        # but they won't be in the correct sort (by votes in the last
-        # 30 days). It is better to return them in the wrong order
-        # than not to return them at all.
-        documents = _db_documents_for(locale, topics, products)
-        cache.add(_documents_for_cache_key(locale, topics, products), documents)
-
-    return documents
-
-
-def _es_documents_for(locale, topics=None, products=None):
-    """ES implementation of documents_for."""
-    s = (
-        DocumentMappingType.search()
-        .values_dict("id", "document_title", "url", "document_parent_id", "document_summary")
-        .filter(
-            document_locale=locale,
-            document_is_archived=False,
-            document_category__in=settings.IA_DEFAULT_CATEGORIES,
+    qs = (
+        Document.objects.filter(
+            locale=locale,
+            is_archived=False,
+            current_revision__isnull=False,
```

Comment:
shouldn't you also check for `is_approved` here?

Reply:
I believe the `current_revision` is always approved:

https://github.com/mozilla/kitsune/blob/3370b8ef85676df11b1dd04b961cb53fe60c1b2f/kitsune/wiki/models.py#L84-L87